system/security/cryptohelper/pke.cpp

```diff
@@ -102,39 +115,69 @@ class CLoadedPublicKeyFromMemory : public CLoadedKey
 
 class CLoadedPrivateKeyFromFile : public CLoadedKey
 {
-public:
-    CLoadedPrivateKeyFromFile(const char *keyFile, const char *passPhrase)
+private:
+    MemoryBuffer passPhraseMB;
+    void load(const char *keyFile, const char *passPhrase)
     {
         if (!loadKeyFromFile(keyFile))
             throwEVPException(0, "loadKeyFromFile: failed to open private key");
-        RSA *rsaKey = PEM_read_bio_RSAPrivateKey(keyBio, nullptr, nullptr, (void*)passPhrase);
+        RSA *rsaKey;
+        if (passPhraseMB.length())
+            rsaKey= PEM_read_bio_RSAPrivateKey(keyBio, nullptr, passphraseCB, (void *)&passPhraseMB);//Binary passphrase
```

Comment:
(void *)&passPhraseMB should be passPhraseMB.bufferBase() instead.

Reply:
I need access to the MemoryBuffer object to query the length, so I need to pass the reference

Reply:
Got it.  I should have read the code better!